client/src/main/java/com/vesoft/nebula/client/storage/StorageClient.java

```diff
@@ -1132,6 +1183,92 @@ private long getEdgeId(String spaceName, String edgeName) {
+            throw new IllegalArgumentException(
+                    "the user,password,graphAddress can not be null,"
+                            + " please config them first by setXXX()");
+        }
+        SyncConnection graphConnection = new SyncConnection();
+        String[] graphAddrAndPort = graphAddresss.split(":");
+        if (graphAddrAndPort.length != 2) {
+            throw new IllegalArgumentException("the graph address is invalid.");
+        }
+        graphConnection.open(new HostAddress(graphAddrAndPort[0].trim(),
+                Integer.valueOf(graphAddrAndPort[1].trim())), timeout);
+        AuthResult authResult = graphConnection.authenticate(user, password);
+        long sessionId = authResult.getSessionId();
+        ResultSet resultSet = new ResultSet(
+                graphConnection.execute(sessionId, "DESC USER " + user),
+                authResult.getTimezoneOffset());
+        if (!resultSet.isSucceeded()) {
+            throw new RuntimeException("get spaces for user " + user + " failed, "
+                    + resultSet.getErrorMessage());
+        }
+        if (resultSet.isEmpty()) {
+            throw new RuntimeException("there's no space for user " + user + " to have permission"
+                    + " to access.");
+        }
+
+        for (int i = 0; i < resultSet.getRows().size(); i++) {
+            List<ValueWrapper> values = resultSet.rowValues(i).values();
+            String role = values.get(0).asString();
+            String space = values.get(1).asString();
+            if (!role.equalsIgnoreCase("BASIC")) {
+                spaceLabelWriteList.put(space, null);
+            } else {
+                List<String> labels = new ArrayList<>();
+                // get the tags and edges that the user has read permission for
+                String showGrants = String.format("USE %s; show grants %s", space, user);
+                ResultSet userGrantResult = new ResultSet(graphConnection.execute(sessionId,
+                        showGrants),
+                        authResult.getTimezoneOffset());
+                if (!userGrantResult.isSucceeded()) {
+                    throw new RuntimeException("get tags for user " + user
+                            + " failed, " + userGrantResult.getErrorMessage());
+                }
+                List<ValueWrapper> readTags = userGrantResult.colValues("READ(TAG)");
+                if (!readTags.isEmpty()) {
+                    for (ValueWrapper v : readTags.get(0).asList()) {
+                        labels.add(v.asString());
+                    }
+                }
+                List<ValueWrapper> readEdges = userGrantResult.colValues("READ(EDGE)");
+                if (!readEdges.isEmpty()) {
+                    for (ValueWrapper v : readEdges.get(0).asList()) {
+                        labels.add(v.asString());
+                    }
+                }
+                spaceLabelWriteList.put(space, labels);
```

Comment:
check if label and edge can have same name or not

Reply:
the tag and edge cannot has the same name for one space.
![image](https://github.com/vesoft-inc/nebula-java/assets/16240361/8c67f3f4-ba79-40d6-b548-da8224dae42a)